### PR TITLE
review

### PR DESCRIPTION
**Pull request by Pol-Gallardo**

### CODE REVIEW

File: src/PerpetualGuardian.sol [hunk 1]
```diff
@@ -31,6 +32,7 @@ contract PerpetualGuardian is ERC4626, Ownable {
     ERC20 public DAI;
     PositionsSummary totalLongPositions;
     PositionsSummary totalShortPositions;
+    AggregatorV3Interface private s_priceFeed;
```
Comment (0xdimo) [inline src/PerpetualGuardian.sol:35]: What do you think about calling it `ethDaiPriceFeed`

File: src/PerpetualGuardian.sol [hunk 2]
```diff
@@ -127,39 +115,30 @@ contract PerpetualGuardian is ERC4626, Ownable {
...
-    //TODO: Replace with price feed
-    uint256 ethPrice = 1_000;
-
-    function getETHPriceInDAI() public view returns (uint256) {
-        return ethPrice;
-    }
-
-    function changeEthPriceInDAI(uint256 newPrice) public {
-        ethPrice = newPrice;
+    function getPositionCollateral(uint256 positionIndex) public view returns (uint256) {
```
Comment (0xdimo) [inline src/PerpetualGuardian.sol:139]: Why do we need this :?

File: test/unit/PerpetualGuardian.t.sol
Comment (0xdimo) [inline test/unit/PerpetualGuardian.t.sol:1]: Please uncomment the tests. We need to have them. Also add tests for the price feed

**Review by 0xdimo — CHANGES_REQUESTED** [review]

File: script/DeployPerpetualGuardian.s.sol [hunk 3]
```diff
@@ -11,13 +11,14 @@ contract DeployPerpetualGuardian is Script {
     function run() external returns (PerpetualGuardian) {
         HelperConfig config = new HelperConfig();
 
-        (ERC20 dai, uint256 deployerKey) = config.activeNetworkConfig();
+        (address ethUsdPriceFeed, address usdc, uint256 deployerKey) = config.activeNetworkConfig();
 
         vm.startBroadcast(deployerKey);
         PerpetualGuardian perpetualGuardian = new PerpetualGuardian(
-            dai,
```
[the remark this answers is not in the document]
Comment (Pol-Gallardo) [reply script/DeployPerpetualGuardian.s.sol:18]: OKey It's fine I like it Dai

File: src/PerpetualGuardian.sol (hunk 1 above)
Comment (Pol-Gallardo) [reply src/PerpetualGuardian.sol:35]: yes better It's more information about this

File: src/PerpetualGuardian.sol (hunk 2 above)
Comment (Pol-Gallardo) [reply src/PerpetualGuardian.sol:139]: This function is to work the increasePositionSize

File: test/unit/PerpetualGuardian.t.sol
Comment (Pol-Gallardo) [reply test/unit/PerpetualGuardian.t.sol:1]: I commented out the tests to be able to do a 'forge build.' I'll rewrite them to make them work but using the price feed and fetching real prices.

File: src/PerpetualGuardian.sol (hunk 2 above)
Comment (0xdimo) [reply src/PerpetualGuardian.sol:139]: You can get the collateral from the mapping

File: test/unit/PerpetualGuardian.t.sol
Comment (0xdimo) [reply test/unit/PerpetualGuardian.t.sol:1]: Great, waiting for this 💯

File: src/PerpetualGuardian.sol (hunk 2 above)
Comment (Pol-Gallardo) [reply src/PerpetualGuardian.sol:139]: Did you see how I modified the logic in 'openPosition' and 'increasePositionSize'?

File: src/PerpetualGuardian.sol [hunk 4]
```diff
@@ -68,6 +79,9 @@ contract PerpetualGuardian is ERC4626, Ownable {
     //--------------------------------
     //TODO: Check if position is healthy
     function openPosition(uint256 size, uint256 collateral, bool isLong) public {
+        if (size > 15) {
+            revert PerpetualGuardian__SizeExceeds();
+        }
```
Comment (0xdimo) [inline src/PerpetualGuardian.sol]: Why do we check the size :?

File: test/unit/PerpetualGuardian.t.sol [hunk 5]
```diff
@@ -107,21 +119,50 @@ contract PerpetualGuardianTest is StdCheats, Test {
         console.log(perpetualGuardian.totalSupply());
     }
 
+    modifier addLP() {
```
Comment (0xdimo) [inline test/unit/PerpetualGuardian.t.sol]: I love this one!

File: test/unit/PerpetualGuardian.t.sol [hunk 6]
```diff
@@ -107,21 +119,50 @@ contract PerpetualGuardianTest is StdCheats, Test {
...
+        perpetualGuardian.addLiquidity(70_000);
+        vm.stopPrank();
+        _;
+    }
+
+    //////////////////
+    // Price Tests //
+    //////////////////
+
+    function testGetTokenAmountFromUsd() public {
```
Comment (0xdimo) [inline test/unit/PerpetualGuardian.t.sol:126]: nice!

File: test/unit/PerpetualGuardian.t.sol [hunk 7]
```diff
@@ -23,14 +30,17 @@ contract PerpetualGuardianTest is StdCheats, Test {
     //--------------------------------
     //SETUP
     //--------------------------------
-    function setUp() public {
+    function setUp() external {
```
Comment (0xdimo) [inline test/unit/PerpetualGuardian.t.sol:33]: why external :?

**Review by 0xdimo — CHANGES_REQUESTED** [review]

File: lib/solmate
Comment (0xdimo) [inline lib/solmate:1]: I don't think we need solmate :?

File: script/HelperConfig.s.sol [hunk 8]
```diff
@@ -3,30 +3,53 @@
...
+        if (block.chainid == 11155111) {
+            activeNetworkConfig = getSepoliaEthConfig();
+        } else {
+            activeNetworkConfig = getOrCreateAnvilEthConfig();
+        }
+    }
+
+    function getSepoliaEthConfig() public view returns (NetworkConfig memory) {
+        return NetworkConfig({
+            ethUsdPriceFeed: 0x694AA1769357215DE4FAC081bf1f309aDC325306,
```
Comment (0xdimo) [inline script/HelperConfig.s.sol:33]: don't we need eth/DAI pricefeed?

File: src/PerpetualGuardian.sol [hunk 9]
```diff
@@ -7,10 +7,19 @@ import {ERC20} from "openzeppelin-contracts/contracts/token/ERC20/ERC20.sol";
...
-import {AggregatorV3Interface} from "chainlink-brownie-contracts/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";
+import {AggregatorV3Interface} from
+    "chainlink-brownie-contracts/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";
 import {PriceConverter} from "./PriceConverter.sol";
 
 contract PerpetualGuardian is ERC4626, Ownable {
+    ///////////////////
+    // Errors
+    ///////////////////
+    error PerpetualGuardian__SizeExceeds();
```
Comment (0xdimo) [inline src/PerpetualGuardian.sol:18]: let's delete this one if we don't use it

File: test/unit/PerpetualGuardian.t.sol [hunk 10]
```diff
@@ -151,32 +189,30 @@ contract PerpetualGuardianTest is StdCheats, Test {
...
         assertEq(perpetualGuardian.totalAssets(), 32_500);
     }
 
-    function testShortPositionPriceDownMovement() public {
-        addLiquidity();
+    function testShortPositionPriceDownMovement() public addLP {
         openTradePosition(trader1, 1_000, 5_000, false);
 
-        perpetualGuardian.changeEthPriceInDAI(500);
+        int256 ethUsdUpdatedPrice = 500e8;
```
Comment (0xdimo) [inline test/unit/PerpetualGuardian.t.sol:214]: dai instead of USD

**Pull request merged**